### ISSUE TITLE
💡[FEATURE]: Add Interactive Gradio Web UI for Brain Tumor Detection Model

### ISSUE THREAD

**Issue by vansh2604-star** [Niketkumardheeryan/ML-CaPsule#2160]
**Is your feature request related to a problem? Please describe.**
I am frustrated when I want to quickly test the brain tumor detection model, but I have to either execute individual Jupyter Notebook code blocks step-by-step or run a full Flask web application locally (which involves managing separate HTML, CSS styles, and route setups). This makes local testing heavier and makes it harder to host the model interactively on platforms like Hugging Face Spaces.

**Describe the solution you'd like**
I would like to add a lightweight, interactive **Gradio** web application (`gradio_app.py`) directly inside the `Brain Tumor Detection` folder.
This app will:
1. Load the pre-trained `best_model.h5` model.
2. Accept MRI scan uploads (PNG/JPEG) through a simple drag-and-drop user interface.
3. Classify the upload and display the probability confidence scores for all 4 classes (`Glioma`, `Meningioma`, `Pituitary`, and `No tumor`) in real-time.
4. Allow users to run the app instantly with a single command: `python gradio_app.py`.

**Describe alternatives you've considered**
1. **Keeping Flask only:** Traditional web servers are harder to containerize or deploy dynamically to lightweight ML hosting platforms (like Hugging Face Spaces) that natively run Streamlit or Gradio.
2. **Jupyter Notebook only:** This requires users to run raw code cells, which is less accessible for non-technical users or beginners looking for a quick demo.

**Additional context**
Implementing Gradio is straightforward and requires very few dependencies. Below is the proposed script structure for `gradio_app.py`:

```python
import os
import numpy as np
import tensorflow as tf
import gradio as gr
from PIL import Image

dir_path = os.path.dirname(os.path.realpath(__file__))
model_path = os.path.join(dir_path, "best_model.h5")
model = tf.keras.models.load_model(model_path)
class_labels = ['Glioma', 'Meningioma', 'No tumor', 'Pituitary']

def predict_tumor(img):
    if img is None:
        return "No image uploaded."
    img = img.resize((224, 224))
    img_array = tf.keras.preprocessing.image.img_to_array(img)
    img_array = np.expand_dims(img_array, axis=0)
    predictions = model.predict(img_array)[0]
    return {class_labels[i]: float(predictions[i]) for i in range(len(class_labels))}

demo = gr.Interface(
    fn=predict_tumor,
    inputs=gr.Image(type="pil", label="Upload Brain MRI Scan"),
    outputs=gr.Label(num_top_classes=4, label="Prediction Confidence"),
    title="Brain Tumor Classification Hub",
    description="Upload an MRI scan to classify it into Glioma, Meningioma, Pituitary, or No Tumor."
)

if __name__ == "__main__":
    demo.launch()
